src/basic/process-util.c

```diff
@@ -2099,7 +2100,28 @@ int posix_spawn_wrapper(
 #if HAVE_PIDFD_SPAWN
         _cleanup_close_ int pidfd = -EBADF;
 
+        /* If we have observed that SPAWN_CGROUP is not supported, remove the flag */
+        if (!setcgroup_supported && FLAGS_SET(flags, POSIX_SPAWN_SETCGROUP)) {
+                flags &= ~POSIX_SPAWN_SETCGROUP;
+                r = posix_spawnattr_setflags(&attr, flags);
+                if (r != 0)
+                        return -r;
+        }
```

Comment:
you can drop this whole block now

Reply:
ah yes, indeed.

I left the check in:
```
+        if (FLAGS_SET(flags, POSIX_SPAWN_SETCGROUP)) {
+                flags &= ~POSIX_SPAWN_SETCGROUP;
+                r = posix_spawnattr_setflags(&attr, flags);
+                if (r != 0)
+                        return -r;
+        }
```

But maybe it can be removed as well?

Reply:
The change should be OK. Let's keep it.